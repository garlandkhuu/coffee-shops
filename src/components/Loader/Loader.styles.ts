import styled, { keyframes } from 'styled-components';

import { LoaderProps } from '../../interfaces';

const spinAnimation = keyframes`
  0% { transform: rotate(0deg); }
  100% { transform: rotate(360deg); }
`;

export const Spinner = styled.div<LoaderProps>`
  width: ${({ size }) => (size ? `${size}px` : '100%')};
  height: ${({ size }) => (size ? `${size}px` : '100&%')};
  border: ${({ theme, loaderWidth }) =>
    `${loaderWidth ? `${loaderWidth}px` : '16px'} solid ${theme.colors.grey}`};
  border-top: ${({ theme, loaderWidth }) =>
    `${loaderWidth ? `${loaderWidth}px` : '16px'} solid ${theme.colors.grey2}`};
  border-radius: 50%;
  animation: ${spinAnimation} 2s linear infinite;
  margin: ${({ margin }) => margin || 0};
`;
