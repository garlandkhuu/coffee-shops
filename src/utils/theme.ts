export interface Theme {
  colors: {
    beigeLight: string;
    brown: string;
    brown2: string;
    black: string;
    blackOpaque: string;
    white: string;
    grey: string;
    grey2: string;
    sage: string;
    red: string;
  };
  fonts: {
    regular: string;
    bold: string;
    decorated: string;
  };
  maxPageWidth: number;
}

const theme: Theme = {
  colors: {
    beigeLight: '#e2d5ca',
    brown: '#492e22',
    brown2: '#a6816a',
    black: '#000000',
    blackOpaque: 'rgba(0, 0, 0, 0.5)',
    white: '#FFFFFF',
    grey: '#dbd8d7',
    grey2: '#969696',
    sage: '#209971',
    red: '#d43d69',
  },
  fonts: {
    regular: 'Lato',
    bold: 'Lato',
    decorated: 'Besley',
  },
  maxPageWidth: 1500,
};

export default theme;
