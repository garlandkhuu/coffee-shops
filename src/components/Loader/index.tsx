import { Spinner } from './Loader.styles';
import { LoaderProps } from '../../interfaces';

const Loader = ({ size, loaderWidth, className, margin }: LoaderProps) => (
  <Spinner
    className={className}
    size={size}
    loaderWidth={loaderWidth}
    margin={margin}
  />
);

export default Loader;
