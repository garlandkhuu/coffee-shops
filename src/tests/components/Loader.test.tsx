import 'jsdom-global/register';
import { Provider } from 'react-redux';
import { ThemeProvider } from 'styled-components';
import configureStore from 'redux-mock-store';
import { shallowToJson } from 'enzyme-to-json';
import 'jest-styled-components';

import Loader from '../../components/Loader';
import theme from '../../utils/theme';
import { shallow } from '../setup/test-setup';

const mockStore = configureStore();

describe('Loader', () => {
  let component: JSX.Element;

  beforeEach(() => {
    component = (
      <ThemeProvider theme={theme}>
        <Provider store={mockStore()}>
          <Loader />
        </Provider>
      </ThemeProvider>
    );
  });

  afterEach(() => {
    jest.resetAllMocks();
  });

  it('should match snapshot', () => {
    const wrapper = shallow(component);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
