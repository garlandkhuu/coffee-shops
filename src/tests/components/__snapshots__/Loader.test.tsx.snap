// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader should match snapshot 1`] = `
<ContextProvider
  value={
    Object {
      "colors": Object {
        "beigeLight": "#e2d5ca",
        "black": "#000000",
        "blackOpaque": "rgba(0, 0, 0, 0.5)",
        "brown": "#492e22",
        "brown2": "#a6816a",
        "grey": "#dbd8d7",
        "grey2": "#969696",
        "red": "#d43d69",
        "sage": "#209971",
        "white": "#FFFFFF",
      },
      "fonts": Object {
        "bold": "Lato",
        "decorated": "Besley",
        "regular": "Lato",
      },
      "maxPageWidth": 1500,
    }
  }
>
  <Provider
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  >
    <Loader />
  </Provider>
</ContextProvider>
`;
